src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.PlatformCompatabilityCheckDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        private const string MinimumOSPlatformAttribute = nameof(MinimumOSPlatformAttribute);
+        private const string ObsoletedInOSPlatformAttribute = nameof(ObsoletedInOSPlatformAttribute);
+        private const string RemovedInOSPlatformAttribute = nameof(RemovedInOSPlatformAttribute);
+        private const string TargetPlatformAttribute = nameof(TargetPlatformAttribute);
+        private const string Windows = nameof(Windows);
+        private const string OSX = nameof(OSX);
+        private const string macOS = nameof(macOS);
+
+        internal static DiagnosticDescriptor MinimumOsRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableAddedMessage,
+                                                                                      DiagnosticCategory.Interoperability,
+                                                                                      RuleLevel.BuildWarning,
+                                                                                      description: s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        internal static DiagnosticDescriptor ObsoleteRule = DiagnosticDescriptorHelper.Create(RuleId,
+                                                                                      s_localizableTitle,
+                                                                                      s_localizableAddedMessage,
+                                                                                      DiagnosticCategory.Interoperability,
+                                                                                      RuleLevel.BuildWarning,
+                                                                                      description: s_localizableObsoleteMessage,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+        internal static DiagnosticDescriptor RemovedRule = DiagnosticDescriptorHelper.Create(RuleId,
```

Comment:
@terrajobst @jeffhandley as i mentioned in the morning the Obsolete and Removed rules are work same as MinimumOS rule, i.e. if Obsolete attribute defined for an API then that API is restricted to only for that Platform. For now, i am gonna leave this implementation as is as we haven no any Obsolete/Removed attributes added into any API. The fix will come with the functionality update for the Unsupported attributes.

Reply:
Thanks for the heads-up, @buyaa-n. For Preview 8, that sounds good to me.  We'll then change the behavior in RC1 with the other changes.